### PR TITLE
[libmupdf] Update mupdf to 1.22.1

### PR DESCRIPTION
**Pull request by simon987**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

The `use-if-instead-of-ifdef-in-writer.patch` was added by me in https://github.com/microsoft/vcpkg/pull/20328, but it has since been committed upstream in https://github.com/ArtifexSoftware/mupdf/commit/305adc250bffaf so I removed it.

### CODE REVIEW

File: ports/libmupdf/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "libmupdf",
-  "version": "1.19.0-rc2",
-  "port-version": 1,
+  "version": "1.22.1",
+  "port-version": 2,
```
Comment (jimwang118) [inline ports/libmupdf/vcpkg.json:4]:
```suggestion
```

File: ports/libmupdf/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
   "description": "a lightweight PDF, XPS, and E-book library",
   "homepage": "https://github.com/ArtifexSoftware/mupdf",
```
Comment (jimwang118) [inline ports/libmupdf/vcpkg.json:6]:
```suggestion
  "homepage": "https://github.com/ArtifexSoftware/mupdf",
  "license": "AGPL-3.0-only",
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Comment by simon987** [comment]
@jimwang118 it seems that after removing the `port-version` line, the `vcpkg/versions/l-/libmupdf.json` file reverted to version 0:

```
"versions": [
    {
      "git-tree": "c1684ff15eed2c86ba792de4e92d21b2ea58c188",
      "version": "1.22.1",
      "port-version": 0
    },
    {
      "git-tree": "4764feb3f9edca039e2e5847eebdeacc06ef2a70",
      "version": "1.22.1",
      "port-version": 2
    },
    {
      "git-tree": "9ca3ac09949f15d339ff747bb36f8593ff087959",
      "version": "1.19.0-rc2",
      "port-version": 1
    },
    ...
```

**Comment by jimwang118** [comment]
You need to re-run the command `./vcpkg x-add-version [portname] --overwrite-version` command to generate the `vcpkg/versions/l-/libmupdf.json` file and CI will end normally after submitting again.

**Comment by simon987** [comment]
> The value is 0. This is correct.

Ah okay!

File: versions/l-/libmupdf.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "c1684ff15eed2c86ba792de4e92d21b2ea58c188",
+      "version": "1.22.1",
+      "port-version": 0
+    },
+    {
+      "git-tree": "4764feb3f9edca039e2e5847eebdeacc06ef2a70",
+      "version": "1.22.1",
+      "port-version": 2
+    },
```
Comment (jimwang118) [inline versions/l-/libmupdf.json:12]:
```suggestion
    },
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Review by jimwang118 — APPROVED** [review]

**Pull request merged**